packages/lms/src/lib/components/navigation/EntityNav.svelte

```diff
@@ -1,60 +1,112 @@
 <script lang="ts">
-	import type { EntityMeta } from '@mollify/types';
+  import type { EntityMeta } from '@mollify/types';
+  import { readable, writable } from 'svelte/store';
 
-	export let entities: Array<EntityMeta> | EntityMeta = [];
-	$: _entities = Array.isArray(entities) ? entities : [entities];
+  export let entities: Array<EntityMeta> | EntityMeta = [];
+  $: _entities = Array.isArray(entities) ? entities : [entities];
 
-	export let open = false;
+  // Use Svelte writable store for the "open" state
+  let openStates = writable({}); // Initialize as a writable store
+  let isMobile = false; // Initialize the mobile state
 
-	function toggle() {
-		open = !open;
-	}
+  // Check if the code is running in a browser environment
+  if (typeof window !== 'undefined') {
+    // Add a listener to resize events to track the mobile state
+    window.addEventListener('resize', () => {
+      isMobile = window.innerWidth <= 768;
+      if (!isMobile) {
+        isMobile = false; // Close the navigation on wider screens
+      }
+    });
+  }
+
+  function toggle(entityId: string) {
+    openStates.update((states) => {
+      states[entityId] = !states[entityId];
+      return { ...states };
+    });
```

Comment:
It might be my project set up wrong - but when i return the states in a console.log i get a object like this: 
`Object { undefined: true }` 

I think it has something with that the entity elements not having a ID in their object? 
Correct me if i am wrong here..

Reply:
Thats correct, there is no id assigned to it in the `getEntityMetaTree` function. 

I am wondering if at some point it might be required that content is assigned an id in the front-matter to make finding related audio files and the likes easier though.